apps/server/src/modules/tool/service/tool-validation.service.spec.ts

```diff
@@ -72,27 +75,39 @@ describe('ToolValidation', () => {
 			});
 		});
 
-		describe('hasDuplicateAttributes', () => {
-			it('should not find duplicate custom parameters', async () => {
-				const externalToolDO: ExternalToolDO = externalToolDOFactory.withCustomParameters(2).build();
-
-				const result: Promise<void> = service.validateCreate(externalToolDO);
-
-				await expect(result).resolves.not.toThrow();
+		describe('hasDuplicateAttributes is called', () => {
+			describe('when duplicates are given', () => {
+				it('should throw when external tool', async () => {
```

Comment:
this sounds like every external tool should fail, which is not the case. instead choose a description that tells you about the thing that is tested here, for example "should fail for two equal parameters", "should fail for different parameteres with same name", "should fail for names that only differ in capitalisation"